Files/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,546 @@
+                    if (matchingNetDrive != null)
+                    {
+                        fontIconSource.Glyph = "\uE8CE";
+                        tabLocationHeader = matchingNetDrive.Text;
+                    }
+                    else
+                    {
+                        try
+                        {
+                            List<System.IO.DriveInfo> drives = System.IO.DriveInfo.GetDrives().ToList();
+                            System.IO.DriveInfo matchingDrive = drives.FirstOrDefault(x => PathNormalization.NormalizePath(currentPath).Contains(PathNormalization.NormalizePath(x.Name)));
+
+                            if (matchingDrive != null)
+                            {
+                                // Go through types and set the icon according to type
+                                string type = GetDriveTypeIcon(matchingDrive);
+                                if (!string.IsNullOrWhiteSpace(type))
+                                {
+                                    fontIconSource.Glyph = type;
+                                }
+                                else
+                                {
+                                    fontIconSource.Glyph = "\xEDA2"; // Drive icon
+                                }
+                            }
+                            else
+                            {
+                                fontIconSource.Glyph = "\xE74E"; // Floppy icon
+                            }
+                        }
+                        catch (Exception)
+                        {
+                            fontIconSource.Glyph = "\xEDA2"; // Fallback
+                        }
+
+                        tabLocationHeader = PathNormalization.NormalizePath(currentPath);
+                    }
+                }
+                else
+                {
+                    fontIconSource.Glyph = "\xE8B7"; // Folder icon
+                    tabLocationHeader = currentPath.TrimEnd(System.IO.Path.DirectorySeparatorChar, System.IO.Path.AltDirectorySeparatorChar).Split('\\', StringSplitOptions.RemoveEmptyEntries).Last();
+
+                    FilesystemResult<StorageFolderWithPath> rootItem = await FilesystemTasks.Wrap(() => DrivesManager.GetRootFromPathAsync(currentPath));
+                    if (rootItem)
+                    {
+                        StorageFolder currentFolder = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFolderFromPathAsync(currentPath, rootItem));
+                        if (currentFolder != null && !string.IsNullOrEmpty(currentFolder.DisplayName))
+                        {
+                            tabLocationHeader = currentFolder.DisplayName;
+                        }
+                    }
+                }
+            }
+
+            return (tabLocationHeader, fontIconSource);
+        }
+
+        public static string GetDriveTypeIcon(System.IO.DriveInfo drive)
```

Comment:
Should this be moved to its own helper? I think it will be useful beyond MainPageViewModel.

Reply:
Will do!